test/query_generator/mutations_and_inputs/filter_input_objects_test.dart

```diff
@@ -0,0 +1,248 @@
+import 'package:artemis/generator/data.dart';
+import 'package:test/test.dart';
+
+import '../../helpers.dart';
+
+void main() {
+  group('On complex input objects', () {
+    test(
+      'Unused input objects will be filtered out',
```

Comment:
Because I've created a new canonical visitor (responsible for retrieving **all** canonical classes: enums, input objects and fragments). However, not every of those canonical objects are used on each one of the queries, so I'm filtering out the unused values while we don't implement a single place to output canonical objects.